### ISSUE TITLE
can't install using makefile in manjaro kde

### ISSUE THREAD

**Issue by silentbirdie** [pez2001/razer_blackwidow_chroma_driver#37]
her is the output of the comand:

::\033[32m COMPILING librazer_chroma.so\033[0m
================================
::\033[32m COMPILING librazer_chroma.so [DEBUG]\033[0m
========================================
::\033[32m COMPILING razer_bcd [SHARED]\033[0m
================================
::\033[32m COMPILING razer_bcd daemon\033[0m
==============================
::\033[32m COMPILING razer_bcd daemon [DEBUG]\033[0m
======================================
/usr/bin/ld: cannot find -lrazer_chroma_debug
collect2: error: ld returned 1 exit status
Makefile:34: recipe for target 'daemon_debug' failed
make[1]: *** [daemon_debug] Error 1
Makefile:26: recipe for target 'razer_daemon' failed
make: *** [razer_daemon] Error 2

**Comment by terrycain** [comment]
Haven't tried compiling it on Fedora but I'm sure some alterations will be needed. Give me a day or two to whip up a test vm and ill look into it

**Comment by silentbirdie** [comment]
Thanks! It's the kde edition of manjaro

**Comment by BenoitMartin2** [comment]
Hi, same to me. I can't install it on Manjaro. Thanks for your kind help mate!
Even tried this => http://pollonamid.de/index.php/en/my-devices/computer/90-how-to-get-the-razer-blackwidow-chroma-backlight-to-work-with-linux
But couldn't make it.

**Comment by pablomenino** [comment]
I have the same problem on Fedora 23.

gcc -L../lib -rdynamic razer_daemon.do razer_daemon_dbus.do razer_daemon_effects.do razer_daemon_libraries.do razer_daemon_parameters.do razer_daemon_render_nodes.do razer_daemon_types.do -lrazer_chroma_debug -lm -ldl -o razer_bcd_debug_shared 

/usr/bin/ld: cannot find -lrazer_chroma_debug


[log.txt](https://github.com/pez2001/razer_blackwidow_chroma_driver/files/46158/log.txt)

**Comment by terrycain** [comment]
There is a issue with the command in the lib/Makefile (and the others that use LIBDBUS) which doesn't set the LIBDBUS flag. Ensure you have dbus headers installed. Also you'll also need the kernel-headers installed for the kernel your running (aka output of `uname -a`)

I have a pull request which fixes the command which should then work for arch. This commit terrycain/razer_blackwidow_chroma_driver@32f607a contains the differences you need to get the Makefiles to work.

**Comment by terrycain** [comment]
BenoitMartin2. Dont post massive errors in pull requests, its just annoying.

What files are in the lib folder after you typed make

**Comment by BenoitMartin2** [comment]
Next time I'll use a paste, sry.
Here are the files in /lib => http://imgur.com/z23g32x

**Comment by terrycain** [comment]
Thanks. cd into the daemon_controller and run `make`, what does that output + what files are there in there

**Comment by BenoitMartin2** [comment]
Same :/   
Complete output: http://pastebin.archlinux.fr/1746199

**Comment by Flennic** [comment]
Hi folks,

I did the changes to the makefiles but I still get the same problem:

http://pastebin.com/nq61YKj5
chown: cannot access ‘/etc/init.d/razer_bcd’: Not a directory

I am not sure if it'd help if I create that directory before doing make?

terrycain can you give a small introduction how to start the driver and how to echo stuff to it (if it finishes compiling one day :-)).

Thanks for all your work!

**Comment by terrycain** [comment]
@BenoitMartin2 make sure you have all the libraries installed as @Fennic said. Otherwise im not sure whats wrong.

@Flennic make install wont work on arch, its designed for debian. There is a daemon start script in `install_files/init.d/razer_bcd_debian`. It wont work out of the box but what it essentially does is unbinds the chroma from the generic-hid driver and binds it to our driver and then starts the daemon.

**Comment by Flennic** [comment]
Just do be curious: I executed ./activate_driver.sh and aht I got is this:

Unbinding device (0003:1532:0203.0003) from hid-generic
Binding device (0003:1532:0203.0003) to razerkbd
Bind Successful

Which looks good in my eyes. If the deamon is running now, is there a way to to send commands to it like make all keys illuminated in blue or smth. like this?

**Comment by terrycain** [comment]
Unless my script is giving you false hope looks like the kernel driver is loaded.

Whats the output of `ls /sys/bus/hid/drivers/razerkbd/`?

Best make sure the driver is actually binded before trying other things

**Comment by Flennic** [comment]
Output is:
ls /sys/bus/hid/drivers/razerkbd/
0003:1532:0203.0003  bind  module  new_id  uevent  unbind

How to check if the driver is actually binded?

Hell I feel like a 4 years old kid asking all those questions but I have no idea how drivers work or how close/far I am from getting it to run. I also found a folder systemd with a razer_bcd.service file. Is it possilbe if I copy all needed files manually to the correct location that it could work?

**Comment by terrycain** [comment]
Run `ls /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/`
The fact that `0003:1532:0203.0003` is there means its binded but knowing the contents makes it easier to explain

That's fine it takes a while to tame the linux kernel and its many drivers.
Yes, essentially if you copy all the required files to the specific places then things will work. If you give me a few days Im starting to populate the wiki to document all of this. (And reduce the question ;) ).

The daemon still needs a lot of work but you can manually operate the driver (once again ill make a wiki page to say what data the device driver files accept).

**Comment by Flennic** [comment]
ls /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/
country  input       mode_breath  mode_reactive  mode_wave          reset           subsystem
driver   macro_keys  mode_custom  mode_spectrum  power              set_brightness  temp_clear_row
hidraw   modalias    mode_none    mode_static    report_descriptor  set_key_row     uevent

The Wiki idea sounds really good. I really appreciate all of your hard work, sadly I can't help you that much. If I get more into this after the Wiki is up, maybe I can do some work. We will see. Just letting you know that I am really thankful.

Another question is: The deamon that already exists, does it work the same way in ubuntu and arch or are there differences? Because you said it still needs a lot of work? I guess just one color-backlight would be awesome too :-)

Take your time to do this and if there is paperwork to do (Wiki or somethink were your could need help), just let me know!

**Comment by terrycain** [comment]
As sudo run this
```
echo "1" >  /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_wave
```
"should" set the keyboard into wave mode.

The wiki should be good once I've done it. The daemon works well its just not very flexible at the moment. The only issue is the differences in the layout of the filesystems between ubuntu and arch, if everything is where the daemon expect it to be it should run.

**Comment by flennic** [comment]
```
sudo echo "1" >  /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_wave
bash: /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_wave: Permission denied
```

No idea why that is. Can't really change the 0 to 1 with vim either, says sync error and that the file has been modified since opening it.

One last question: I can see the ubuntu location for all of the files in the makefile. Do you have any idea where the files for systemd go? Like /etc/init.d/racer_bcd where I have to put that? Becasue as far as I can read the makefile is works well until the point where it's trying to create /etc/init.d/ files and symlinks.

If we get to the point where everything is working as intended (yeah I know :D), is it hard to make a pacman package for the AUR, do you have any idea how much work that is? But anyways, that's not on the list right now. But if I am bored, I am willing to try it. Could help a lot of people.

**Comment by terrycain** [comment]
Run it as root
sudo su
then the echo thing.

Once I get the wiki up and running ill document everything, then feel free to make an AUR package and update the documentation to getting it working on arch

**Comment by Flennic** [comment]
```
[root@pc razer_blackwidow_chroma_driver]# echo "1" >  /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_wave
```

Works now, but nothing's happening :/

**Comment by terrycain** [comment]
What about
```
echo "1" >  /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_spectrum
```

**Comment by Flennic** [comment]
```
echo "1" >  /sys/bus/hid/drivers/razerkbd/0003:1532:0203.0003/mode_spectrum
```
Same behaviour. Works but nothing happens.

**Comment by terrycain** [comment]
Yeh the mode_static takes hexadecimal. \xFF\xFF\x0 should be yellow.

I'm going to write a page about the driver files. Then about what goes where on the filesystem and then about the daemon.

It "migh" just work if you run `GIT_CHECKOUT/daemon/razer_bcd -fv` `GIT_CHECKOUT` being the path to the folder you cloned. If it runs and doesn't exit straight away, start typing into a text editor and see if the daemon effect is working.

**Comment by Flennic** [comment]
Hey there, hexadecimal works.

But therefore I have another strange thing: When I start my computer and booting into arch, the keyboard mode is set so spectrum. So somehow the driver is started instantly, even before decrypting my harddrive.
After booting I have to active the driver, so that the 0003:... file appears and after that I can sent instructions to the keyboard again. But I can't "disable" colors. What worked is setting static mode, but I was unable to stop the spectrum mode.

How can the keybord be illuminated even before the OS booted? I thought there is no daemon running? :-)

**Comment by dubbleohnexus** [comment]
The keyboard illumination is at a hardware level, it'll turn on the
backlights as soon as its given power. Evidence of this is if you were to
use the keyboard in your bios, it would light up.

**Comment by flennic** [comment]
Okay, seems like I do not fully understand how this works. After my PC starts, which part sends the data to the keyboard? You say it's on hardware level, so which hardware is doing that? As far as I know the keyboard itself is stupid (unlike Logitech's keyboards) and is not able to save any information itself.

Or is it possible it's still powered via USB and can keep the data that way?

**Comment by terrycain** [comment]
No the keyboard is not stupid. There are a few effects that the keyboard can perform itself, the driver only allows us to tell the keyboard "do this effect".

The keyboard knows how to do spectrum,wave,reactive,static,none and breathing effects

**Comment by dubbleohnexus** [comment]
I'm not real sure if the keyboard can save what backlighting mode to run,
but the LEDs will turn on as soon as they're given power. The
microcontroller in the keyboard must have a small amount of onboard memory
(either flash or RAM) to store backlighting profiles, otherwise you
wouldn't be able to use them at all.

**Comment by Flennic** [comment]
Thanks for all the information. I configured the keyboard on Mac OS X before and when I plugged it into my Linux machine it didn't show any light effects. Therefore I thought the wrong way.

If you plan to develop an opensource driver, how do you know how to communicate with the keyboard? It's not like Razer is telling us how. But anyways, thats not part of this issue.

**Comment by terrycain** [comment]
Back before razer went all cloud sync, there used to be ways to store stuff on the keyboard. The keyboard can probably still do that but to find that out I'd need to find an ancient version of the razer tool.

As for how we communicate with the keyboard, we find out what makes it tick by reverse engineering the USB traffic. Then using that knowledge we send stuff to the keyboard.

**Comment by BenoitMartin2** [comment]
Hi,

I'm still stucked. Have all libraries mentionned by @Flennic. BTW, as told earlier, already tried the "Arch" package from pollonamid.
http://pastebin.archlinux.fr/1752329 => ln: impossible de créer le lien symbolique « /etc/rc2.d/S24razer_bcd »: Aucun fichier ou dossier de ce type

**Comment by terrycain** [comment]
@BenoitMartin2  run `ls -lR` inside the repository and put it on pastebin

**Comment by BenoitMartin2** [comment]
Sorry but which one do you want I ls?

**Comment by terrycain** [comment]
which ever folder you ran install-driver.sh in

**Comment by BenoitMartin2** [comment]
Here they are:
 - Pez2001: http://pastebin.archlinux.fr/1752340
 - Pollonamid: http://pastebin.archlinux.fr/1752367

**Comment by terrycain** [comment]
We dont support anything in the Pollonamid repo, thats the job for whoever forked that (but from looking at it, it compiled sucessfully).

As for the Pez2001 repo, run `sudo make clean` in that and then give me the output of `make`

**Comment by BenoitMartin2** [comment]
Okay. Here's the make output: http://pastebin.archlinux.fr/1752368

**Comment by terrycain** [comment]
That looked like its working. Output of `ls -lR`

**Comment by BenoitMartin2** [comment]
http://pastebin.archlinux.fr/1752389

**Comment by terrycain** [comment]
Yeah its compiled fine. The `make install` most likely wont work as its not designed for arch. Look at the wiki for this repository, and google, you need to move the razerkbd.ko to the appropriate place to "install" the driver.

**Comment by Flennic** [comment]
Just to clarify things: The makefile didn't work for me. What I did was:
From the downloaded git repo:
* cd install_files/init.d/
* sudo ./activate_driver.sh
* ls /sys/bus/hid/drivers/razerkbd/
* Look for a device name like 0003:1532:0203.0003
* cd into that folder
* there you will find various files. Look inside the wiki what you have to echo in which file to use the driver
* keep in mind, sudo does not work here, you have to switch to root with sudo su

**Comment by BenoitMartin2** [comment]
Thank you @terrycain, it seems it's okay : http://pastebin.archlinux.fr/1752410
I'm going on with hid-generic

**Comment by terrycain** [comment]
@Flennic to clarify, the makefile didn't install things but it did compile the code as otherwise nothing would of worked ;)

@BenoitMartin2 Cool

**Comment by Flennic** [comment]
Oh yes I thought thats for sure. Sorry I didn't mention :-)

**Comment by BenoitMartin2** [comment]
It's working!  Thanks a lot :)  Now... I just need to understand how to deal with it and bind it a better way. The wave's gonna drive me mad :/

@terrycain Do you have a BTC wallet or else?

**Comment by terrycain** [comment]
@BenoitMartin2 Agreed, the wave looks cool until you actually want to use the keyboard ;)
I have paypal (terry@terrys-home.co.uk) but never really got into bitcoins.

Once you have it binded there is another page on how to change colours (the static works but some things are waiting on some of my changed to get merged)

**Comment by Flennic** [comment]
Another issue is that the numblock icon looks diffuse. Any idea why is that or how to fix that?

**Comment by terrycain** [comment]
@Flennic you mean its a little blurry??

**Comment by Flennic** [comment]
Yes :-)

**Comment by BenoitMartin2** [comment]
Have a beer (or more) then @terrycain ;)  I read your Wiki, I'll now have to configure my keyboard. Thanks again gyuz!

**Comment by terrycain** [comment]
@Flennic That's just the way the case is made. Its the same on my Ultimate 2013 too, their new Ultimate 2016 [Link](http://www.razerzone.com/gb-en/store/razer-blackwidow-ultimate-2016) has changed that I think. My mates ordered that so it shall be supported soon ;)

@BenoitMartin2 Cheers that will go towards beer and towards my razer mamba which shall be added as soon as it arrives ;)

**Comment by BenoitMartin2** [comment]
Do you know if there's anyway to get their Windows config tool work with Linux? Cause despite your work/help made me able to use my keyboard a bit better, it's still really not easy to change colors as we code or play X/Y game.

**Comment by terrycain** [comment]
@BenoitMartin2 the windows tool wont work, you could run it in wine but it still wouldn't know how to use the driver. If you can get the daemon working there is a simple python app which lets you do a few things.

There are a few scripts to use the driver etc... Its a work in progress but its getting there

**Comment by Flennic** [comment]
If there is a way to save things on teh keyboard itself, that would be great. Then you could configure it using a Windows or Mac OS X Machine and use it with windows. That's what I was doing with my previous keyboards from Logitech. But I don't have that much hope with this Razer keyboard. Still letting in glow in one color of my choice is already okay :-)

**Comment by BenoitMartin2** [comment]
Wall I remember on my AlienWare laptop I could configure the lights using Windows before having it to Linux. And the keyboard remains colored. This might be possible with this Razer too. I'll git it a try.

**Comment by terrycain** [comment]
I might look into getting the razer cloud sync to work. The keyboards used to have profiles but not sure if that'll work anymore.

The keyboard "should" remember the last setting that the razer synapse tool set but im not sure if it realises this once its powered off and on

**Comment by dubbleohnexus** [comment]
After unplugging and replugging my keyboard it remembers the last setting
it was on. I dunno about rebooting the computer though, I'll have to try
that later.

**Comment by terrycain** [comment]
@Flennic Cheers, more beer for me ;)

**Comment by silentbirdie** [comment]
can someone explain to me how to install it as this is all going over my head and I do not understand it.

**Comment by dubbleohnexus** [comment]
I thought the wiki did a pretty decent job of explaining it. Pretty much
just c/p the commands from there and do what it says and it'll work.

**Comment by Flennic** [comment]
Hi everbody,

I'm trying to get into systemd and I want to get it to work under Arch Linux. As far as I can see the makefile first creates all the binarys and then copies them to the right location. This is the following part:
```
	cp install_files/init.d/razer_bcd /etc/init.d
	chown root:root /etc/init.d/razer_bcd
	cp install_files/share/bash_keyboard_functions.sh /usr/share/razer_bcd/bash_keyboard_functions.sh
	chown root:root /usr/share/razer_bcd/bash_keyboard_functions.sh
	ln -fs ../init.d/razer_bcd /etc/rc2.d/S24razer_bcd
	ln -fs ../init.d/razer_bcd /etc/rc3.d/S24razer_bcd
	ln -fs ../init.d/razer_bcd /etc/rc4.d/S24razer_bcd
	ln -fs ../init.d/razer_bcd /etc/rc5.d/S24razer_bcd

	ln -fs ../init.d/razer_bcd /etc/rc0.d/K02razer_bcd
	ln -fs ../init.d/razer_bcd /etc/rc1.d/K02razer_bcd
	ln -fs ../init.d/razer_bcd /etc/rc6.d/K02razer_bcd
```
In Arch Linux we do not have /etc/init.d - therefore we got systemd. Following the guide I mentioned earlier I need to copy the systemd daemon file to
```
/etc/systemd/system/abrtd.service
```
Fortunately there already is a systemd file which was already designed for Ubuntu/Debian, it looks like this:
```
[Unit]
Description=Razer Keyboard Daemon
After=network.target auditd.service

[Service]
EnvironmentFile=-/etc/default/razer_bcd
ExecStartPre=/usr/share/razer_bcd/systemd_helpers.sh bind
ExecStart=/usr/sbin/razer_bcd --pid-file /var/run/razer_bcd.pid
ExecStopPost=/usr/share/razer_bcd/systemd_helpers.sh unbind
KillMode=process
TimeoutStopSec=30
Type=forking
PIDFile=/var/run/razer_bcd.pid

[Install]
WantedBy=multi-user.target
Alias=razer_bcd.service
```
Following the path's given there shows me, that in /usr/share/racer_bcd is no systemd_helpers.sh, instead there is a folder called fx and instide this one there is:
```
ls /usr/share/razer_bcd/fx/
pez2001_collection.so   pez2001_light_blast.so  pez2001_progress_bar.so
pez2001_compute_fft.so  pez2001_mixer.so
```

I think there is not that much missing to get ther daemon work unter Arch Linux, I guess it's just path adjustments. But I'm really not that much into that, but willing to learn and to do that.
Therefore my question is: Does anyone has the knowledge to do this and is willing to help me? Maybe via mail or something like this. I really want to get the daemon working!

**Comment by terrycain** [comment]
@Flennic the ./pacakge_for_debian script uses the systemd stuff.

Move the following to `/usr/share/razer_bcd`
`install_files/share/systemd_helpers.sh`
`install_files/share/bash_keyboard_functions.sh`

Can you get the driver to work on its own? It would be best getting that working before making it a service. As root try `razer_bcd -f`

**Comment by silentbirdie** [comment]
@dubbleohnexus doesn't explain what the kernel compile thing is or how to do it and it confuses me I'm still new to installing drivers like this it is my first time.

**Comment by terrycain** [comment]
@silentbirdie Remove the repository. Clone it again. Make the patches I referenced above and then run `make` and put the output on pastebin.

If any of you have a Razer Mamba well I have just got a few colours working plus battery/charging numbers too (I know a lot more info but just haven't coded it yet).

**Comment by Flennic** [comment]
Hello everybody,

didn't have that much time the last days.
@terrycain I did as you said, output is:
```
[root@pc share]# razer_bcd -f
Starting razer blackwidow chroma daemon
```

**Comment by terrycain** [comment]
@Flennic Did it exit immediately or did it just sit there?

**Comment by Flennic** [comment]
```
[root@pc share]# razer_bcd -f
Starting razer blackwidow chroma daemon
[root@pc share]# 
```

After one second the second root@ appeared. Does this answer your question?

**Comment by terrycain** [comment]
Yeah that answers it. Is the driver binded?

Running that as you did it should just sit there and wait and not return another prompt until killed.

**Comment by Flennic** [comment]
> Is the driver binded?
Yes I guess so. My keyboard is illuminated blue all the time. Any way to test if the driver is binded?

> Running that as you did it should just sit there and wait and not return another prompt until killed.
What exactly do you mean with another prompt until killed? Sorry for my disability :(

**Comment by terrycain** [comment]
Returning a prompt, meaning the second `[root@pc share]# ` would not appear until you ended the razer_bcd program (normally with `Ctrl+C`).

**Comment by Flennic** [comment]
Ah okay now I get the point. I didn't use Ctrl+C. It gave me the prompt back without any interaction of mine. So it means, it is NOT working as intended, right?

**Comment by terrycain** [comment]
Correct. 
Is the driver binded?

If the driver is binded then paste the output of `strace razer_bcd -f` on pastebin

**Comment by Flennic** [comment]
http://pastebin.com/L7h3HcrF

**Comment by terrycain** [comment]
in the repo, inside the daemon folder is there a file called `razer_bcd_debug_static`? If so run that `razer_bcd_debug_static -f -v` and give me the output

**Comment by Flennic** [comment]
```
[root@pc daemon]# ./razer_bcd_debug_static 
Starting razer blackwidow chroma daemon as a daemon
killing razer_bcd parent process
[root@pc daemon]# ./razer_bcd_debug_static -f -v
Starting razer blackwidow chroma daemon in the foreground
opening chroma lib
error no compatible device found
razer_bcd: error initializing daemon
```

Keep in mind I just restarted. So I guess the driver is unbinded. Shall I bind it again for this test?

**Comment by terrycain** [comment]
Yes. Driver must be binded for this to work

**Comment by Flennic** [comment]
See edit last post.

**Comment by terrycain** [comment]
Ok I know what's happening. I'm a bit busy at the moment, I'll reply tonight/tomorrow but all hope is not lost ;).

**Comment by Flennic** [comment]
Allright :-) Take your time and thanks for all your efforts! Good night!

**Comment by terrycain** [comment]
Try rerunning `razer_bcd -f` now that the driver is bound.

If that doesn't work then run `razer_bcd_debug_static -f -v` but run it from the root of the repository (not from inside the daemon folder), like this `./daemon/razer_bcd_debug_static -f -v`

And of course post output

**Comment by Flennic** [comment]
```
[user@pc init.d]$ sudo razer_bcd -f
Starting razer blackwidow chroma daemon
```
Gives me back my lsd keyboard. Stops instantly -> gives me the command prompt back.
 ```
user@pc razer_blackwidow_chroma_driver]$ sudo ./daemon/razer_bcd_debug_static -f -v
Starting razer blackwidow chroma daemon in the foreground
opening chroma lib
loading fx lib: daemon/fx/pez2001_collection_debug.so
registered effect: First One (uid:1)
registered effect: Breathing Color (uid:2)
registered effect: Random Color Waves (uid:3)
registered effect: Random Color Waves #2 (uid:4)
registered effect: Random Color Waves #3 (uid:5)
registered effect: Random Color Waves #4 (uid:6)
registered effect: Copper Bars #4 (uid:7)
registered effect: Arrayi Bars (uid:8)
registered effect: Arrayi Bars 2 (uid:9)
registered mix effect: Default Mixer (uid:10)
loading fx lib: daemon/fx/pez2001_light_blast_debug.so
registered effect: Lightblaster (uid:11)
 (Blast.1 ## opacity:1.000000 / -1,-1)                                          ft:9ms,fps:111.111115(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1) (Compute::KeyRingBuffer.1 ## )                                                                  ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:49ms,fps:20.408163(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:49ms,fps:20.408163(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                        ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:49ms,fps:20.408163(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:51ms,fps:19.607843(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                     ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1) (Compute::KeyRingBuffer.1 ## )      ft:50ms,fps:20.000000(20) (Blast.1 ## opacity:1.000000 / -1,-1)                                                            ^C
[user@pc razer_blackwidow_chroma_driver]$ ^C
```
Second try from the root directory looks fine.

**Comment by terrycain** [comment]
OK I am am assuming razer_bcd is installed in /usr/sbin, whats the output of `ldd /usr/sbin/razer_bcd`.

What I think is the issue is that the main razer_bcd is looking for files which don't exist then just dies.
also make sure `/usr/share/razer_bcd/fx` has these files
```
$ ls -l /usr/share/razer_bcd/fx
total 60K
-rwxrwxr-x 1 terry terry  22K Nov 28 01:12 pez2001_collection.so*
-rwxrwxr-x 1 terry terry 9.8K Nov 28 01:12 pez2001_light_blast.so*
-rwxrwxr-x 1 terry terry  14K Nov 28 01:12 pez2001_mixer.so*
-rwxrwxr-x 1 terry terry 5.5K Nov 28 01:12 pez2001_progress_bar.so*
```

One big issue which I might tackle at some point is that the daemon has no logging which isn't very helpful when your trying to troubleshoot things.

**Comment by Flennic** [comment]
```
[user@pc 0003:1532:0203.0003]$ ldd /usr/sbin/razer_bcd
	linux-vdso.so.1 (0x00007fff137a8000)
	libdbus-1.so.3 => /usr/lib/libdbus-1.so.3 (0x00007f342f207000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f342ef09000)
	libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007f342ec06000)
	libSDL2_image-2.0.so.0 => /usr/lib/libSDL2_image-2.0.so.0 (0x00007f342e9e0000)
	librazer_chroma.so => /usr/lib/librazer_chroma.so (0x00007f342e7dc000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f342e438000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f342e21b000)
	libsystemd.so.0 => /usr/lib/libsystemd.so.0 (0x00007f342f5bd000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f342e017000)
	librt.so.1 => /usr/lib/librt.so.1 (0x00007f342de0f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f342f458000)
	libcap.so.2 => /usr/lib/libcap.so.2 (0x00007f342dc0b000)
	libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007f342d9f4000)
	liblzma.so.5 => /usr/lib/liblzma.so.5 (0x00007f342d7ce000)
	liblz4.so.1 => /usr/lib/liblz4.so.1 (0x00007f342d5bc000)
	libgcrypt.so.20 => /usr/lib/libgcrypt.so.20 (0x00007f342d2da000)
	libgpg-error.so.0 => /usr/lib/libgpg-error.so.0 (0x00007f342d0c7000)
	libattr.so.1 => /usr/lib/libattr.so.1 (0x00007f342cec2000)
[user@pc 0003:1532:0203.0003]$ 
```

```
[user@pc 0003:1532:0203.0003]$ ls -l /usr/share/razer_bcd/fx
total 56
-rwxr-xr-x 1 root root 18792 28. Nov 20:14 pez2001_collection.so
-rwxr-xr-x 1 root root  4520 28. Nov 20:14 pez2001_compute_fft.so
-rwxr-xr-x 1 root root  6864 28. Nov 20:14 pez2001_light_blast.so
-rwxr-xr-x 1 root root 10952 28. Nov 20:14 pez2001_mixer.so
-rwxr-xr-x 1 root root  4520 28. Nov 20:14 pez2001_progress_bar.so
```

**Comment by terrycain** [comment]
Most strange. Make sure librazer_chroma.so is up-to-date. I would remove `/usr/sbin/razer_bcd` `/usr/lib/librazer_chroma.so` `/usr/share/razer_bcd` and reinstall all of them and try again

**Comment by BenoitMartin2** [comment]
Hi, good new :)

I installed Windows so as to play a bit AND to test my keyboard. The loaded profile on Windows remains activated switching to Linux. I use a KVM switch.
Didn't reboot yet the Linux PC, I'll get back to you later bout that.

**Comment by dubbleohnexus** [comment]
I can confirm that saved profile remains across reboots, PC switches,
driver not being started, and anything else you can imagine.

**Comment by BenoitMartin2** [comment]
Nice then! I'll go set my game profiles etc.

**Comment by typh00n666** [comment]
hi, I can't install on ubuntu 14.04 LTS, getting this error:

razer_chroma.o: In function `razer_create_device':
razer_chroma.c:(.text+0x12b1): undefined reference to `__stack_chk_fail_local'
razer_chroma.o: In function `razer_poll_input_device':
razer_chroma.c:(.text+0x1f90): undefined reference to `__stack_chk_fail_local'
razer_chroma.o: In function `razer_device_set_game_mode':
razer_chroma.c:(.text+0x3d81): undefined reference to `__stack_chk_fail_local'
razer_chroma.o: In function `razer_device_set_wave_mode':
razer_chroma.c:(.text+0x415c): undefined reference to `__stack_chk_fail_local'
razer_chroma.o: In function `razer_device_set_brightness':
razer_chroma.c:(.text+0x43aa): undefined reference to `__stack_chk_fail_local'
/usr/bin/ld: librazer_chroma.so: hidden symbol `__stack_chk_fail_local' isn't defined
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make[1]: *** [razer_chroma_shared] Error 1
make: *** [librazer_chroma] Error 2

**Comment by terrycain** [comment]
Path to the current directory? Command used to make

**Comment by typh00n666** [comment]
I don't understand. can you help me install this driver personally? can you contact me?

**Comment by terrycain** [comment]
What is the filepath to where you checked out the repository? And what command did you run to get that output above.

**Comment by darkphoenixff4** [comment]
I'm going to post it here, because I'm betting it's related, but I can't compile on openSUSE 42.1 Leap.

`\n::\033[32m COMPILING razer_bcd controller [SHARED]\033[0m
===========================================
gcc -L../lib -rdynamic razer_daemon_controller.o -lrazer_chroma -lrazer_chroma_controller -lm -ldl `pkg-config --libs dbus-1`  -o razer_bcd_controller
gcc  `pkg-config --cflags dbus-1` -DUSE_DBUS -Wl,-E -Wall -Wextra -W -w -Os -std=c99 -DMAJOR_VERSION=0 -DMINOR_VERSION=4 -fPIC -DPIC -DLINUX -D_XOPEN_SOURCE=700 -c -o razer_daemon_dbus.o razer_daemon_dbus.c
/usr/lib64/gcc/x86_64-suse-linux/5/../../../../x86_64-suse-linux/bin/ld: cannot find -lrazer_chroma
/usr/lib64/gcc/x86_64-suse-linux/5/../../../../x86_64-suse-linux/bin/ld: cannot find -lrazer_chroma_controller
collect2: error: ld returned 1 exit status
Makefile:49: recipe for target 'daemon_controller_shared' failed
make[1]: *** [daemon_controller_shared] Error 1
Makefile:66: recipe for target 'daemon_controller' failed
make: *** [daemon_controller] Error 2`

Also, I have a Blackwidow, a Firefly, an Orbweaver, a Deathstalker and a Kraken 7.1, so if you need any possible debugging info from any of those, I'll be happy to contribute.

**Comment by terrycain** [comment]
Ugh this problem appears every now and again. What's the contents of the `lib` directory in the repository?

Oooh lots of devices, yes once this works we'll get to work supporting all of them ;)

**Comment by terrycain** [comment]
Also full output of `make` command put in pastebin.com would be helpful

**Comment by JKenyon1** [comment]
Unable to install on Fedora 24 getting the following:

make -C /lib/modules/4.5.5-300.fc24.x86_64/build SUBDIRS=/home/jasonk/pez2001-razer_chroma_drivers-88eaeed/driver modules > /dev/null 2>&1
Makefile:19: recipe for target 'driver' failed
make: *** [driver] Error 2

**Comment by terrycain** [comment]
Can you please recreate the ticket on [here](https://github.com/terrycain/razer_drivers).

**Comment by GabMus** [comment]
In the new repo that terrycain just posted there are PKGBUILDs for Arch based distros. They should work no problem in manjaro, too
